src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -315,14 +315,18 @@ class FastSync(
       if (checkHeadersChain(headers)) {
         processHeaders(peer, headers) match {
           case ParentDifficultyNotFound(header) =>
-            log.debug("Parent difficulty not found for block {}, not processing rest of headers", header.number)
-            processSyncing()
+            // We could end in wrong fork and get blocked so we should rewind our state a little
+            // we blacklist peer just in case we got malicious peer which would send us bad blocks, forcing us to rollback
+            // to genesis
+            log.info("Parent difficulty not found for block {}, not processing rest of headers", header.idTag)
```

Comment:
maybe all this logs that end up in a blacklisting action could be used `.warn` instead of `info`, in that way we could easily search for them. WDYT?